usr/sbin/regionsrv-enabler-azure

```diff
@@ -97,20 +97,27 @@ utils.start_logging()
 service_name = 'guestregister'
 license_type = get_license_type()
 
-if ('BYOS' in license_type and has_license_changed(license_type) and
-    not utils.uses_rmt_as_scc_proxy()):
+if ('BYOS' in license_type and not utils.uses_rmt_as_scc_proxy()):
+    # Registered to SCC
+    license_has_changed = has_license_changed(license_type)
```

Comment:
The assignment can be moved outside the if condition then we do not have to make a second function call in the `else:` clause. Meaning in the case where there is no state file we also save ourselves one stack unwind do to the exception.

Reply:
I do not agree, 
- if license type includes BYOS and not RMT as SCC proxy -> we check the license
- else if it does not include BYOS and there is an update server -> we do not check the license

Unless I'm missing a path or workflow of the code (in the current PR), we do not always check the license ?
If we do not mind doing it outside even in the case where license type does not include BYOS
and system has an SMT server, fine by me

There is another option, moving the check from
```python
if ('BYOS' in license_type and not utils.uses_rmt_as_scc_proxy()):
    license_has_changed = has_license_changed(license_type)
    if (
        license_has_changed or
        (update_server and utils.is_registered(update_server.get_FQDN()))
    ):
        ...
```
to
```python
license_has_changed = has_license_changed(license_type)
if ('BYOS' in license_type and
    not utils.uses_rmt_as_scc_proxy() and
    (license_has_changed or 
    (update_server and utils.is_registered(update_server.get_FQDN()))
):
    ...
```

Reply:
Correct I missed the nested ` if update_server:` in the `else:` condition.